internal/proxy/proxy.go

```diff
@@ -307,8 +316,7 @@ func (p *Proxy) loginJenkins(pci ProxyCacheItem, osoToken string) (int, []*http.
 	} else {
 		log.WithField("ns", pci.NS).Infof("Accessing Jenkins route %s", jenkinsURL)
 	}
-	c := http.DefaultClient
-	resp, err := c.Do(req)
+	resp, err := p.httpClient.Do(req)
```

Comment:
So when this request times out after 5 seconds, you get and err, right?

Reply:
I think so

Reply:
Can we not test 'loginJenkins' in isolation and exercise the two main cases - once returning with success and once timing out? At least that would give some confidence in this method.